src/VisualStudio/Core/Def/Implementation/Workspace/VisualStudioDocumentNavigationService.cs

```diff
@@ -246,6 +257,37 @@ public bool TryNavigateToPosition(Workspace workspace, DocumentId documentId, in
             }
         }
 
+        private bool TryNavigateToMappedFile(Workspace workspace, ISpanMappingService spanMappingService, Document generatedDocument, TextSpan textSpan)
+        {
+            var vsWorkspace = workspace as VisualStudioWorkspaceImpl;
+            Contract.ThrowIfNull(vsWorkspace);
+            var results = ThreadingContext.JoinableTaskFactory.Run(async () =>
+            {
+                return await spanMappingService.MapSpansAsync(generatedDocument, SpecializedCollections.SingletonEnumerable(textSpan), CancellationToken.None).ConfigureAwait(true);
+            });
+
+            if (!results.IsDefaultOrEmpty)
+            {
+                var mappedSpanResult = results.Single();
+                // TODO - Move to IOpenDocumentService - https://github.com/dotnet/roslyn/issues/45954
+                vsWorkspace.OpenDocumentFromPath(mappedSpanResult.FilePath, generatedDocument.Project.Id);
```

Comment:
It's a bit funky we're passing this Project.Id here -- at this point it's a file that isn't necessarily a related to that project context, is it important that we're still passing something?